apps/test-suite/tests/FileSystemNext.ts

```diff
@@ -260,5 +261,26 @@ export async function test({ describe, expect, it, ...t }) {
         expect(() => src.move(dst)).toThrow();
       });
     });
+
+    describe('Downloads files', () => {
+      it('downloads a file to a target file', async () => {
+        const url = 'https://httpbin.org/image/jpeg';
+        const file = new File(testDirectory + 'image.jpeg');
+        const output = await download(url, file);
+        expect(file.exists()).toBe(true);
+        expect(output.path).toBe(file.path);
+      });
+
+      it('downloads a file to a target directory', async () => {
+        const url = 'https://httpbin.org/image/jpeg';
+        const directory = new Directory(testDirectory);
+        const output = await download(url, directory);
+        const file = new File(
+          testDirectory + (Platform.OS === 'android' ? 'jpeg.jpg' : 'jpeg.jpeg')
```

Comment:
Can't we use the same extension on all platforms?

Reply:
We rely on OS functions to get a filename based on URL headers – I don't want to write my own now, but can look into that a bit more in the future.